### ISSUE TITLE
Cant extend array or object configuration via environment variables

### ISSUE THREAD

**Issue by desa** [influxdata/kapacitor#1344]
Currently users can only modify existing configuration elements.

The following test will panic:
```
func TestConfig_Parse_EnvOverrideAddArray(t *testing.T) {
    // Parse configuration.
    var c server.Config
    if _, err := toml.Decode(`
[[influxdb]]
urls=["http://localhost:8086"]
`, &c); err != nil {
        t.Fatal(err)
    }

    if err := os.Setenv("KAPACITOR_INFLUXDB_0_URLS_0", "http://localhost:18086"); err != nil {
        t.Fatalf("failed to set env var: %v", err)
    }

    if err := os.Setenv("KAPACITOR_INFLUXDB_1_URLS_0", "http://localhost:7076"); err != nil {
        t.Fatalf("failed to set env var: %v", err)
    }

    if err := c.ApplyEnvOverrides(); err != nil {
        t.Fatalf("failed to apply env overrides: %v", err)
    }

       if c.InfluxDB[0].URLs[0] != "http://localhost:18086" {
        t.Fatalf("unexpected url 0: %s", c.InfluxDB[0].URLs[0])
    } else if c.InfluxDB[1].URLs[0] != "http://localhost:7076" {
        t.Fatalf("unexpected url 0: %s", c.InfluxDB[0].URLs[0])
    }
}
```

Users should be able to add additional configuration to Kapacitor through environment variables.

**Comment by sofuture** [comment]
Yep, investigated a bit and this causes #1280 (which shouldn't have been closed). Also see [this](https://community.influxdata.com/t/unable-to-set-smtp-to-via-environment-variable/917). I will work on a fix today.

**Comment by sofuture** [comment]
Turns out it's a bit messy to do right :) I'm working on it though, maybe not highest priority since I can workaround the part that's biting me.

**Comment by desa** [comment]
@sofuture awesome to hear that you're working on this. If there's anything we can do to help, let us know.

**Comment by rdxmb** [comment]
Will there be a fix for that?

**Comment by itzg** [comment]
A solution for this would be great since I am unable to configure the kafka event handler brokers via an environment variable `KAPACITOR_KAFKA_0_BROKERS_0`.

**Comment by xpunch** [comment]
I think we can add a specific key as slice length, then apply child element through prefix_index_field.
`
if len, err := strconv.Atoi(os.Getenv(fmt.Sprintf("%s_LEN", prefix))); err == nil && len > 0 {
	t := reflect.MakeSlice(reflect.SliceOf(element.Type().Elem()), len, len)
	for i := 0; i < len; i++ {
		f := t.Index(i)
		if err := c.applyEnvOverrides(fmt.Sprintf("%s_%d", prefix, i), f, structKey); err != nil {
			return err
		}
	}
	element.Set(t)
	break
}
`
